lib/chef/user.rb

```diff
@@ -144,12 +287,17 @@ class << self
     end
 
     def self.list(inflate=false)
-      response = Chef::REST.new(Chef::Config[:chef_server_url]).get_rest('users')
-      users = if response.is_a?(Array)
```

Comment:
Just added this commit for backwards compatibility support: https://github.com/chef/chef/commit/55834783ee26a53cfad113433a57c691b3cbc114

I also just wrote up my thoughts and what I actually changed in at the top of this PR. Please read and let me know what you think.